azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -117,17 +143,55 @@ public int createImageType(ImageType imageType) {
 
   @Override
   public List<ImageType> getAllImageTypes() throws ImageMgmtException {
-    FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
     try {
-      return databaseOperator
+      return this.databaseOperator
           .query(FetchImageTypeHandler.FETCH_ALL_IMAGE_TYPES, fetchImageTypeHandler, true);
-    } catch (SQLException ex) {
+    } catch (final SQLException ex) {
       log.error(FetchImageTypeHandler.FETCH_ALL_IMAGE_TYPES + " failed.", ex);
       throw new ImageMgmtDaoException(
           "Unable to fetch all image type metadata ");
     }
   }
 
+  @Override
+  public List<ImageType> getAllImageTypesWithOwnerships() throws ImageMgmtException {
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+    List<ImageType> imageTypes = new ArrayList<>();
+    try {
+      imageTypes = this.databaseOperator
+          .query(FetchImageTypeHandler.FETCH_ALL_IMAGE_TYPES, fetchImageTypeHandler, true);
+      for (final ImageType imageType : imageTypes) {
+        imageType.setOwnerships(getImageOwnerships(imageType.getName()));
+      }
+    } catch (final SQLException ex) {
+      log.error(FetchImageTypeHandler.FETCH_ALL_IMAGE_TYPES + " failed.", ex);
+      throw new ImageMgmtDaoException(
+          "Unable to fetch all image type metadata ");
+    }
+    return imageTypes;
+  }
+
+  /**
+   * Gets ownership metadata based on image type name.
+   *
+   * @param imageTypeName
+   * @return List<ImageOwnership>
+   */
+  private List<ImageOwnership> getImageOwnerships(final String imageTypeName) {
+    final FetchImageOwnershipHandler fetchImageOwnershipHandler = new FetchImageOwnershipHandler();
+    try {
+      return this.databaseOperator
+          .query(FetchImageOwnershipHandler.FETCH_IMAGE_OWNERSHIP_BY_IMAGE_TYPE_NAME,
+              fetchImageOwnershipHandler, imageTypeName);
+    } catch (final SQLException ex) {
+      log.error(FetchImageOwnershipHandler.FETCH_IMAGE_OWNERSHIP_BY_IMAGE_TYPE_NAME + " failed.",
+          ex);
+      throw new ImageMgmtDaoException(
+          "Unable to fetch image ownership for image type : " + imageTypeName);
```

Comment:
Nit: Unable to fetch ownership for image type:

Reply:
Done.